auth-api/src/auth_api/models/task.py

```diff
@@ -49,25 +49,27 @@ class Task(BaseModel):
     user = relationship('User', foreign_keys=[related_to], lazy='select')
 
     @classmethod
-    def fetch_tasks(cls, task_type: str, task_status: List[str],  # pylint:disable=too-many-arguments
-                    task_relationship_status: str,
-                    page: int, limit: int):
+    def fetch_tasks(cls, task_search: TaskSearch):
         """Fetch all tasks."""
         query = db.session.query(Task)
+        if task_search.name:
+            query = query.filter(Task.name.ilike(f'%{task_search.name}%'))
+        if task_search.type:
+            query = query.filter(Task.type == task_search.type)
+        if task_search.status:
+            query = query.filter(Task.status.in_(task_search.status))
+        if task_search.date_submitted:
+            query = query.filter(Task.date_submitted == task_search.date_submitted)
 
-        if task_type:
-            query = query.filter(Task.type == task_type)
-        if task_status:
-            query = query.filter(Task.status.in_(task_status))
-        if task_relationship_status:
-            if task_relationship_status == TaskRelationshipStatus.PENDING_STAFF_REVIEW.value:
-                query = query.filter(Task.relationship_status == task_relationship_status).order_by(
+        if task_search.relationship_status:
```

Comment:
I think if `task_search.relationship_status == TaskRelationshipStatus.PENDING_STAFF_REVIEW.value` should be sufficient

Reply:
Also this seems redundant.. 

doing filter to the same value in both branches.. 

Easier to move the order_by out to it's own conditional branch and turf the if else..

Reply:
done.

Reply:
Doesn't look done after your last push